### PR TITLE
FindAllByField accept arrays now

### PR DESCRIPTION
**Pull request by danmartuszewski**
Hi,
I've found that it could be useful to be able to use findAllByX() with array as first parameter.
What do you think?

### CODE REVIEW

File: src/ORM/Table.php [hunk 1]
```diff
@@ -1787,7 +1787,13 @@ protected function _dynamicFinder($method, $args)
                 ));
             }
             foreach ($fields as $field) {
-                $conditions[$this->aliasField($field)] = array_shift($args);
+                $value = array_shift($args);
+
+                if (is_array($value)) {
+                    $field .= ' IN';
```
Comment (markstory) [inline src/ORM/Table.php]: This is dangerous as if an end user controls the values they can manipulate queries into finding multiple rows when the application is expecting a single row.

**Comment by dmartuszewski** [comment]
Not exactly cause it's only for findAllByX not findById for example

**Comment by AD7six** [comment]
Not a fan of this, except for simple usage `findByX` methods should imo be discouraged. The assumption that this change is not dangerous because it's only for `findAllByX` has implicit assumptions that I don't think are true.

**Comment by markstory** [comment]
@dmartuszewski Fair point, but I suspect people use these methods in the following way

```php
$query = $this->Users->findByUserName($this->request->data('username'));
foreach ($query as $row) {
  $this->updatePassword($row);
}
```

With your change the post data could be manipulated into an array and find multiple users vs. finding a single one.

**Comment by ionas** [comment]
Just was curious about how rails is doing that lately (as I encountered those finders in Rails 0.9 first time)
Refs: https://github.com/rails/activerecord-deprecated_finders
E.g. find_**all**_by_x / find**All**ByX is deprecated in rails4, as well.

**Comment by markstory** [comment]
Closing as I'm not comfortable with the possibility of allowing request data to potentially manipulate queries from `=` into `IN` queries.

**Pull request closed without merging**